tests/integration/test_commands.py

```diff
@@ -1,20 +1,965 @@
+
+    def test_submit_test_manifest(self, runner: CliRunner) -> None:
+        """Tests for a successful submission"""
+        # commented out, this command causes an error
+        # https://sagebionetworks.jira.com/browse/SCHEMATIC-126
+        """
+        result = runner.invoke(
+            model,
+            [
+                "--config",
+                "config_example.yml",
+                "submit",
+                "-mp",
+                "tests/data/mock_manifests/CLI_tests/CLI_biospecimen.csv",
+                "-vc",
+                "Biospecimen",
+                "-mrt",
+                "table_and_file",
+                "-d",
+                "syn23643250"
+            ],
+        )
+        assert result.exit_code == 0
+        """
+
+
+class TestValidateCommand:
+    """Tests the schematic/models/commands validate command"""
+
+    def test_validate_valid_manifest(self, runner: CliRunner) -> None:
+        """Tests for validation with no errors"""
+        result = runner.invoke(
+            model,
+            [
+                "--config",
+                "config_example.yml",
+                "validate",
+                "--manifest_path",
+                "tests/data/mock_manifests/Valid_Test_Manifest.csv",
+                "--data_type",
+                "MockComponent",
+                "--project_scope",
+                "syn54126707",
+            ],
+        )
+        assert result.exit_code == 0
+        assert result.output.split("\n")[4] == (
+            "Your manifest has been validated successfully. "
+            "There are no errors in your manifest, "
+            "and it can be submitted without any modifications."
+        )
+
+    def test_validate_invalid_manifest(self, runner: CliRunner) -> None:
+        """Tests for validation with no errors"""
+        result = runner.invoke(
+            model,
+            [
+                "--config",
+                "config_example.yml",
+                "validate",
+                "--manifest_path",
+                "tests/data/mock_manifests/CLI_tests/CLI_patient_invalid.csv",
+                "--data_type",
+                "Patient",
+            ],
+        )
+        assert result.exit_code == 0
+        assert result.output.split("\n")[3] == (
+            "error: For attribute Family History in row 2 it does not appear "
+            "as if you provided a comma delimited string. Please check your entry "
+            "('Random'') and try again."
+        )
+        assert result.output.split("\n")[4].startswith("error: 'Random' is not one of")
+        assert result.output.split("\n")[5].startswith("error: 'Random' is not one of")
+        assert result.output.split("\n")[6].startswith("[['2', 'Family History',")
+
+
+class TestManifestCommand:
+    """Tests the schematic/manifest/commands validate manifest command"""
+
+    def test_generate_empty_csv_manifest(self, runner: CliRunner) -> None:
+        """Generate two empty csv manifests"""
```

Comment:
I'll leave it to @BryanFauble for final review, but did we want to follow similar paradigm as the API testing documentation to list exactly what is being tested?

also are there any manual things that need to be checked in the manifests here?

Reply:
Yep it was a lot fo C+P and the modifying for the CLI tests hwere needed. If there's anythign I missed that I should include please let me know :)

Reply:
C+P? I'll leave it to your responsibility to ensure that you've captured everything that the manual test plan outlined.

As for the documentation, I'd be great if there was heavier documentation in the doc strings like:

```
        Download a manifest from the Schematic API and verify that it is a valid Excel
        file. We are validating the following:
        - The first row of the Excel file contains the column headers
        - The first row is locked on scroll
        - Each cell in the first row has a comment/tooltip "TBD"
        - The "Family History" cell has a comment/tooltip with the expected message
        - The cell corresponding to "Sex" in Sheet1 has a dropdown list with values from Sheet2!B2:B4
        - The cell corresponding to "Diagnosis" in Sheet1 has a dropdown list with values from Sheet2!D2:D3
        - The cell corresponding to "Cancer Type" in Sheet1 has a dropdown list with values from Sheet2!F2:F6
        - The workbook contains two sheets: "Sheet1" and "Sheet2"
        - "Sheet2" is hidden
        - The values in "Sheet2" are as expected
        - The fill colors of the first row cells are as expected
        - Conditional formatting is functioning as expected
       
        Manual verification steps:
        - Open the Excel file prefixed with TestManifestGeneration_test_single_manifest_generation_excel.xlsx
        - When Diagnosis = "Cancer", [Cancer Type, Family History] is Light Blue (Required)
```

If there are manul verification steps required for the manifest itself it should follow the same paradigm that Bryan set up

Reply:
+1 My hope is that the docstrings are the single source of great documentation that we can add to the pdocs site.